File: android/src/main/kotlin/com/gdelataillade/alarm/models/VoiceTagSettings.kt
package com.gdelataillade.alarm.models

import com.gdelataillade.alarm.generated.VoiceTagSettingsWire
import kotlinx.serialization.Serializable

@Serializable
data class VoiceTagSettings(
    val enable: Boolean,
    val text: String,
    val volume: Double,
    val speechRate: Double,
    val pitch: Double,
    val loop: Boolean,
    val loopInterval: Long,
) {
    companion object {
        fun fromWire(e: VoiceTagSettingsWire): VoiceTagSettings {
            return VoiceTagSettings(
                e.enable,
                e.text,
                e.volume,
                e.speechRate,
                e.pitch,
                e.loop,
                e.loopInterval,
            )
        }
    }
}

File: ios/Classes/models/VoiceTagSettings.swift
import Foundation

struct VoiceTagSettings: Codable {
    let enable: Bool
    let text: String
    let volume: Double
    let speechRate: Double
    let pitch: Double
    let loop: Bool
    let loopInterval: Int64

    static func from(wire: VoiceTagSettingsWire) -> VoiceTagSettings {
        return VoiceTagSettings(
            enable: wire.enable,
            text: wire.text,
            volume: wire.volume,
            speechRate: wire.speechRate,
            pitch: wire.pitch,
            loop: wire.loop,
            loopInterval: wire.loopInterval
        )
    }
}
